File: cafe-management-backend/src/main/kotlin/de/dkh/cafemanagementbackend/service/BillService.kt
package de.dkh.cafemanagementbackend.service

import de.dkh.cafemanagementbackend.wrapper.BillWrapper
import org.springframework.http.ResponseEntity

interface BillService {
    fun generateBill(requestMap: Map<String, Any>): ResponseEntity<String>
    fun getBills(): ResponseEntity<List<BillWrapper>>
    fun getBillDocument(requestMap: Map<String, String>): ResponseEntity<ByteArray>

}

File: cafe-management-frontend/src/app/services/bill.service.ts
import {Injectable} from '@angular/core';
import {RequestService} from "./extended/request.service";
import {HttpClient} from "@angular/common/http";
import {GlobalConstants} from "../shared/global-constants";
import {Observable} from "rxjs";

@Injectable({
  providedIn: 'root'
})
export class BillService extends RequestService {

  constructor(httpClient: HttpClient) {
    super(httpClient);
  }

  generateBill(data: any) {
    return super.post("/" + GlobalConstants.billPath + "/generate", data);
  }

  getBillDocument(data: any) {
    return super.postForFile("/" + GlobalConstants.billPath + "/getBillDocument", data);
  }

  getBills() {
    return super.get("/" + GlobalConstants.billPath + "/get");
  }


}
